src/App.tsx

```diff
@@ -1,56 +1,82 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { MemoizedDropdown as Dropdown } from './Components/Dropdown/Dropdown';
+
+enum PersonInfo {
+  noSelected = 'No selected person',
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | string>(
+    PersonInfo.noSelected,
+  );
+  const [searchName, setSearchName] = useState('');
+  const [searchQueryName, setSearchQueryName] = useState('');
+
+  const showDropdown = typeof selectedPerson !== 'object'
+    && searchQueryName.length > 0;
```

Comment:
If make null by default your chak will be
```suggestion
  const showDropdown = selectedPerson && searchQueryName.length > 0;
```

Reply:
`!selectedPerson`